File: enc.py
#!/usr/bin/env python

import yaml
import os
import sys
import re

# basic error checking
if len(sys.argv) < 2:
    print "No host provided!"
    sys.exit(0)

node = sys.argv[1]

if node is None:
    print "No host provided!"
    sys.exit(0)

machine_types = [os.path.splitext(m)[0] for m in os.listdir(os.path.join(os.path.dirname(os.path.realpath(__file__)), "data", "machines"))]

try:
    template_type = re.search('^[a-z][a-z]*', node).group(0)
    if template_type not in machine_types:
        template_type = "generic"
except AttributeError:
    template_type = "generic"

try:
    hosts = open(os.path.join(os.path.dirname(os.path.realpath(__file__)), "data", "hosts.yaml"))
    data = yaml.load(hosts)
    print yaml.dump(data[template_type], explicit_start=True, default_flow_style=False).rstrip()
except:
    print "Something went wrong! Are you sure the yaml config exists?"

File: enc.rb
#!/usr/bin/env ruby
# This script was borrowed from carl@hcs (github: @zenzen) 

require 'yaml'

# I would have done this with Hiera's `hiera_include`, but this is more
# flexible, and plus it allows us to pass interesting things to Hiera later.

node = ARGV[0]
raise "No host provided" if node.nil?

machine_types = Dir.entries(File.join(File.dirname(__FILE__), 'data', 'machines')).map { |file| File.basename(file, File.extname(file)) }

hosts = YAML.load_file(File.join(File.dirname(__FILE__), 'data', 'hosts.yaml'))

# For our purposes, use regex to pull out the server type from the hostname
node = node[/^[a-z][a-z]*/,0]
# Default to generic if machine type is not found
node = "generic" if not machine_types.include?(node)

print YAML.dump(hosts.fetch(node))
